### PR TITLE
Fix offboarding template (#4798)

### LINKED ISSUE #4798: Fix offboarding template
Expand both these items:
> Removed the system administrator role from the offboarded employee/contractor's accounts or the offboarded employee/contractor did not perform the system administrator role
> Assigned the system administrator role to the replacement nominated by the project manager or the offboarded employee/contractor was not the sole system administrator

Remove the TODO:
> Removed the GitLab maintainer role from the offboarded employee/contractor (TODO: Fill in details of steps required to do this)

Remove from `Organization administrator actions`:
> Removed the offboarded employee/contractor's Google workspace account and the Google service account deleted above from the azul-dev group in Terra

### PR DESCRIPTION
**Pull request by dsotirho-ucsc**
Connected issues: #4798

### Author

- [x] PR is a draft
- [x] Target branch is `develop`
- [x] Name of PR branch matches `issues/<GitHub handle of author>/<issue#>-<slug>`
- [x] PR title references all connected issues
- [x] PR title matches<sup>1</sup> that of a connected issue <sub>or comment in PR explains why they're different</sub>
- [x] For each connected issue, there is at least one commit whose title references that issue
- [x] PR is connected to all connected issues via ZenHub
- [x] PR description links to connected issues
- [x] Added `partial` label to PR <sub>or this PR completely resolves all connected issues</sub>

<sup>1</sup> when the issue title describes a problem, the corresponding PR
title is `Fix: ` followed by the issue title

### Author (reindex, API changes)

- [x] Added `r` tag to commit title <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label to PR <sub>or this PR does not require reindexing</sub>
- [x] Added `a` (compatible changes) or `A` (incompatible ones) tag to commit title <sub>or this PR does not modify the Azul service API</sub>
- [x] Added `API` label to connected issues <sub>or this PR does not modify the Azul service API</sub>

### Author (chains)

- [x] This PR is blocked by previous PR in the chain <sub>or this PR is not chained to another PR</sub>
- [x] Added `base` label to the blocking PR <sub>or this PR is not chained to another PR</sub>
- [x] Added `chained` label to this PR <sub>or this PR is not chained to another PR</sub>

### Author (upgrading)

- [x] Documented upgrading of deployments in UPGRADING.rst <sub>or this PR does not require upgrading</sub>
- [x] Added `u` tag to commit title <sub>or this PR does not require upgrading</sub>
- [x] Added `upgrade` label to PR <sub>or this PR does not require upgrading</sub>

### Author (operator tasks)

- [x] Added checklist items for additional operator tasks <sub>or this PR does not require additional tasks</sub>

### Author (hotfixes)

- [x] Added `F` tag to main commit title <sub>or this PR does not include permanent fix for a temporary hotfix</sub>
- [x] Reverted the temporary hotfixes for any connected issues <sub>or the `prod` branch has no temporary hotfixes for any connected issues</sub>

### Author (before every review)

- [x] Rebased PR branch on `develop`, squashed old fixups
- [x] Ran `make requirements_update` <sub>or this PR does not touch requirements*.txt, common.mk, Makefile and Dockerfile</sub>
- [x] Added `R` tag to commit title <sub>or this PR does not touch requirements*.txt</sub>
- [x] Added `reqs` label to PR <sub>or this PR does not touch requirements*.txt</sub>
- [x] `make integration_test` passes in personal deployment <sub>or this PR does not touch functionality that could break the IT</sub>

### Peer reviewer (after requesting changes)

Uncheck the *Author (before every review)* checklists.

### Peer reviewer (after approval)

- [x] PR is not a draft
- [x] Ticket is in *Review requested* column
- [x] Requested review from primary reviewer
- [x] Assigned PR to primary reviewer

### Primary reviewer (after requesting changes)

Uncheck the *before every review* checklists. Update the `N reviews` label.

### Primary reviewer (after approval)

- [x] Actually approved the PR
- [x] Labeled connected issues as `demo` or `no demo`
- [x] Commented on connected issues about demo expectations <sub>or all connected issues are labeled `no demo`</sub>
- [x] Decided if PR can be labeled `no sandbox`
- [x] PR title is appropriate as title of merge commit
- [x] `N reviews` label is accurate
- [x] Moved ticket to *Approved* column
- [x] Assigned PR to current operator

### Operator (before pushing merge the commit)

- [x] Checked `reindex` label and `r` commit title tag
- [x] Checked that demo expectations are clear <sub>or all connected issues are labeled `no demo`</sub>
- [x] Squashed PR branch and rebased onto `develop`
- [x] Sanity-checked history
- [x] Pushed PR branch to GitHub
- [x] Pushed PR branch to GitLab `dev` and added `sandbox` label <sub>or PR is labeled `no sandbox`</sub>
- [x] Pushed PR branch to GitLab `anvildev` <sub>or PR is labeled `no sandbox`</sub>
- [x] Build passes in `sandbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [x] Build passes in `anvilbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [x] Reviewed build logs for anomalies in `sandbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [x] Reviewed build logs for anomalies in `anvilbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [x] Deleted unreferenced indices in `sandbox` <sub>or this PR does not remove catalogs or otherwise causes unreferenced indices </sub>
- [x] Deleted unreferenced indices in `anvilbox` <sub>or this PR does not remove catalogs or otherwise causes unreferenced indices </sub>
- [x] Started reindex in `sandbox` <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Started reindex in `anvilbox` <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Checked for failures in `sandbox` <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Checked for failures in `anvilbox` <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Added PR reference to merge commit title
- [x] Collected commit title tags in merge commit title
- [x] Moved connected issues to Merged column in ZenHub
- [x] Pushed merge commit to GitHub

### Operator (after pushing the merge commit)

- [x] Shortened the PR chain <sub>or this PR is not labeled `base`</sub>
- [x] Pushed merge commit to GitLab `dev` <sub>or PR is labeled `no sandbox`</sub>
- [x] Pushed merge commit to GitLab `anvildev` <sub>or PR is labeled `no sandbox`</sub>
- [x] Build passes on GitLab `dev`<sup>1</sup>
- [x] Reviewed build logs for anomalies on GitLab `dev`<sup>1</sup>
- [x] Build passes on GitLab `anvildev`<sup>1</sup>
- [x] Reviewed build logs for anomalies on GitLab `anvildev`<sup>1</sup>
- [x] Deleted PR branch from GitHub
- [x] Deleted PR branch from GitLab `dev`
- [x] Deleted PR branch from GitLab `anvildev`

<sup>1</sup> When pushing the merge commit is skipped due to the PR being
labelled `no sandbox`, the next build triggered by a PR whose merge commit *is*
pushed determines this checklist item.

### Operator (reindex)

- [x] Deleted unreferenced indices in `dev` <sub>or this PR does not remove catalogs or otherwise causes unreferenced indices </sub>
- [x] Deleted unreferenced indices in `anvildev` <sub>or this PR does not remove catalogs or otherwise causes unreferenced indices </sub>
- [x] Started reindex in `dev` <sub>or this PR does not require reindexing</sub>
- [x] Started reindex in `anvildev` <sub>or this PR does not require reindexing</sub>
- [x] Checked for and triaged indexing failures in `dev` <sub>or this PR does not require reindexing</sub>
- [x] Checked for and triaged indexing failures in `anvildev` <sub>or this PR does not require reindexing</sub>
- [x] Emptied fail queues in `dev` deployment <sub>or this PR does not require reindexing</sub>
- [x] Emptied fail queues in `anvildev` deployment <sub>or this PR does not require reindexing</sub>

### Operator

- [x] Unassigned PR

## Shorthand for review comments

- `L` line is too long
- `W` line wrapping is wrong
- `Q` bad quotes
- `F` other formatting problem

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/offboarding.md [hunk 1]
```diff
@@ -41,20 +41,36 @@ ticket has been completed. At that time, put the ticket back to triage.
...
+- [ ] Added the replacement system administrator's Google workspace account to the `azul-anvil-prod` group in Terra <sub>or the offboarded employee/contractor was not the sole system administrator</sub>
+- [ ] Set the `Can manage users (admin)` checkbox for the replacement system administrator in the `azul-dev` group in Terra <sub>or the offboarded employee/contractor was not the sole system administrator</sub>
+- [ ] Set the `Can manage users (admin)` checkbox for the replacement system administrator in the `azul-prod` group in Terra <sub>or the offboarded employee/contractor was not the sole system administrator</sub>
+- [ ] Set the `Can manage users (admin)` checkbox for the replacement system administrator in the `azul-anvil-dev` group in Terra <sub>or the offboarded employee/contractor was not the sole system administrator</sub>
+- [ ] Set the `Can manage users (admin)` checkbox for the replacement system administrator in the `azul-anvil-prod` group in Terra <sub>or the offboarded employee/contractor was not the sole system administrator</sub>
```
Comment (achave11-ucsc) [inline .github/ISSUE_TEMPLATE/offboarding.md]: The [document](https://docs.google.com/document/d/1Kg0dMZmCw6gtkvabD2jYWPZO2Mx_wsC8BJPKdeKTfg0/edit#bookmark=id.fbwffo4y28rv) specifies that contractors never perform organization/system administrator role, which makes this descriptions inconsistent. Perhaps having a different `contractors` section where system/organization administrator actions are excluded is desired?

**Review by achave11-ucsc — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/offboarding.md (hunk 1 above)
Comment (dsotirho-ucsc) [reply .github/ISSUE_TEMPLATE/offboarding.md]: An earlier version of this checklist had optional sections based on the role the offboarded person had, however this complicated the checklist and made it hard to follow, especially with the limited formatting & indentation options available via markdown in a GitHub issue.

Instead, like our PR checklists, it was decided to go with a linear list that could be completed top to bottom with every checkbox being processed.

To address the point you raised that "contractors never perform organization/system administrator role", I've re-worded the `<sub>` notes so that they don't imply a contractor was performing the system admin role.

File: .github/ISSUE_TEMPLATE/offboarding.md [hunk 2]
```diff
@@ -35,26 +35,42 @@ ticket has been completed. At that time, put the ticket back to triage.
...
+- [ ] Assigned the replacement system administrator's account as a maintainer of the `Azul Operators` team on GitHub <sub>or the system administrator is not being replaced</sub>
+- [ ] Handed over credentials of all GitLab root accounts to the replacement system administrator <sub>or the system administrator is not being replaced</sub>
+- [ ] Created a new `dev` GitLab EC2 instance using a private key from the replacement system administrator <sub>or the system administrator is not being replaced</sub>
+- [ ] Created a new `prod` GitLab EC2 instance using a private key from the replacement system administrator <sub>or the system administrator is not being replaced</sub>
+- [ ] Created a new `anvildev` GitLab EC2 instance using a private key from the replacement system administrator <sub>or the system administrator is not being replaced</sub>
+- [ ] Created a new `anvilprod` GitLab EC2 instance using a private key from the replacement system administrator <sub>or the system administrator is not being replaced</sub>
```
Comment (achave11-ucsc) [inline .github/ISSUE_TEMPLATE/offboarding.md]: I don't think this is accurate. Aren't private keys supposed to be private?
The document reads "A _system administrator_ changes the credentials of all GitLab root accounts.", assuming this is the step this checklist items are covering. But I might be reading this wrong, I differ to your expertise.

File: .github/ISSUE_TEMPLATE/offboarding.md [hunk 3]
```diff
@@ -72,33 +88,47 @@ ticket has been completed. At that time, put the ticket back to triage.
...
 - [ ] Removed all role assignments from the offboarded employee/contractor in the `prod` instance of GitLab
 - [ ] Removed all role assignments from the offboarded employee/contractor in the `anvildev` instance of GitLab
 - [ ] Removed all role assignments from the offboarded employee/contractor in the `anvilprod` instance of GitLab
-- [ ] Blocked the GitLab account owned by the offboarded employee/contractor on every GitLab instance in the system
-- [ ] Removed the Google Service accounts owned by the offboarded employee/contractor in the `platform-hca-dev` Google Cloud project
-- [ ] Removed the Google Service accounts owned by the offboarded employee/contractor in the `platform-hca-prod` Google Cloud project
-- [ ] Removed the Google Service accounts owned by the offboarded employee/contractor in the `platform-anvil-dev` Google Cloud project
-- [ ] Removed the Google Service accounts owned by the offboarded employee/contractor in the `platform-anvil-prod` Google Cloud project
-- [ ] Performed an account review and disabled any accounts owned by the offboarded employee/contractor
```
Comment (achave11-ucsc) [inline .github/ISSUE_TEMPLATE/offboarding.md:94]: How is this item in the document being addressed, it seems like it went away from the checklist but is part of the security architecture document.

File: .github/ISSUE_TEMPLATE/offboarding.md [hunk 4]
```diff
@@ -72,33 +88,47 @@ ticket has been completed. At that time, put the ticket back to triage.
...
-- [ ] Performed a resource inventory and deleted any resources owned by the offboarded employee/contractor
```
Comment (achave11-ucsc) [inline .github/ISSUE_TEMPLATE/offboarding.md:95]: Same for this item.

**Review by achave11-ucsc — COMMENTED** [review]
What checklist items cover "The replacement changes the credentials to all AWS root user accounts" for organization admin?

File: .github/ISSUE_TEMPLATE/offboarding.md (hunk 2 above)
Comment (dsotirho-ucsc) [reply .github/ISSUE_TEMPLATE/offboarding.md]: The [GitLab Own](https://docs.google.com/document/d/1Kg0dMZmCw6gtkvabD2jYWPZO2Mx_wsC8BJPKdeKTfg0/edit#heading=h.ohg7b2fljfxq) role section of the [UCSC Data Browser Security Architecture](https://docs.google.com/document/d/1Kg0dMZmCw6gtkvabD2jYWPZO2Mx_wsC8BJPKdeKTfg0/edit) document has further details about removal of this role.

> Removal of this role from an account requires the creation of a new GitLab EC2 instance, using the private key of another system administrator who will then assign this role to themselves as described in the previous section.

You are correct however that this should be an action taken by the replacement system administrator and not the offboarded employee.

File: .github/ISSUE_TEMPLATE/offboarding.md (hunk 3 above)
Comment (dsotirho-ucsc) [reply .github/ISSUE_TEMPLATE/offboarding.md:94]: The PR checklist is indented replace the need for an account review.
The removal/disabling of the various account types are covered in the sections listed below:

- UCSC account: Covered under `ITS actions`
- Google workspace account: Covered under `System administrator actions`
- Google mail account: Covered under `ITS actions` with the disabling the USCS account
- Google service account: Covered under `System administrator actions`
- Terra user account: Covered under `System administrator actions`
- GitHub user account: Covered under `System administrator actions` with the removal of the user from all GitHub organizations
- GitHub machine account: Coverage added to `Offboarded employee/contractor actions` and `System administrator actions`
- GitLab user account: Covered under `System administrator actions`
- GitLab root account: Covered under `System administrator actions` with the creation of new GitLab EC2 instances

File: .github/ISSUE_TEMPLATE/offboarding.md (hunk 4 above)
Comment (dsotirho-ucsc) [reply .github/ISSUE_TEMPLATE/offboarding.md:95]: > What checklist items cover "The replacement changes the credentials to all AWS root user accounts" for organization admin?

I can't find a record of the decision however it is my understanding that the offboarding (and replacement) of an Organization administrator is outside the scope of this checklist, whose purpose is to offboard a person from team Boardwalk.

**Review by achave11-ucsc — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/offboarding.md (hunk 3 above)
Comment (achave11-ucsc) [reply .github/ISSUE_TEMPLATE/offboarding.md:94]: Ah, makes sense. Thank you for the detailed list. ☺️

File: .github/ISSUE_TEMPLATE/offboarding.md (hunk 4 above)
Comment (hannes-ucsc) [reply .github/ISSUE_TEMPLATE/offboarding.md:95]: > the offboarding (and replacement) of an Organization administrator is outside the scope of this checklist

that's my understanding, too

**Review by hannes-ucsc — APPROVED** [review]

**Comment by dsotirho-ucsc** [comment]
Checkbox `Build passes on GitLab dev1` was unchecked. This PR branch and merge commit were not pushed to GitLab (`no sandbox`), however a later push of `develop` to GitLab `dev` https://gitlab.dev.singlecell.gi.ucsc.edu/ucsc/azul/-/pipelines/20372 and GitLab `anvildev` https://gitlab.anvil.gi.ucsc.edu/ucsc/azul/-/pipelines/3757 verifies the success of the build.
